[angle] Improve UX for unofficial CMake package

friendlyanon opened the pull request:
This PR adds a config script to the unofficial CMake package that properly propagates transitive dependencies and usage info for the unofficial CMake package to avoid vcpkg barfing almost all the targets in it.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

friendlyanon committed:
[angle] Propagate transitive dependencies

JonLiu1993 commented:
@friendlyanon, Thanks for your pr, please add `"license": "BSD-3-Clause"` in vcplg.json file.

friendlyanon committed:
[angle] Add usage info for unofficial CMake targets

Only the unofficial::angle::libGLESv2 and unofficial::angle::libEGL
targets are meant to be used, because they are the targets that no other
depends on.

JonLiu1993 commented:
Test usage succeeds with angle:x64-windows triplet.
